### PR TITLE
feat(offline-sqs): Add SQS FIFO support

### PR DESCRIPTION
**Pull request by Letsch22**
Adds support for SQS FIFO queues by picking up the `FifoQueue` and `ContentBasedDeduplication` properties from `serverless.yml` when the queues are automatically created. Addresses #73

### CODE REVIEW

**Comment by AdrieanKhisbe** [comment]
@Letsch22 
There seems to have still some issue with FIFO support from the local providing container.

Don't have time to dive in now, if you have some insight that might help :)

**Comment by Letsch22** [comment]
@AdrieanKhisbe I fixed the issues you were seeing (they were all issues with the tests themselves) in dcc1115.
Per the [SQS FIFO docs](https://docs.aws.amazon.com/AWSSimpleQueueService/latest/SQSDeveloperGuide/FIFO-queues.html), the name of a FIFO queue must always end in `.fifo` (though the resource name doesn't need to follow this convention). Additionally, when you send a message to a FIFO queue you must always include a `MessageGroupId`. I fixed both of these issues in the tests, looks like everything passes now.

File: tests/serverless-plugins-integration/serverless.sqs.autocreate.yml [hunk 1]
```diff
@@ -23,7 +23,7 @@ functions:
               - AutocreatedQueue
               - Arn
       - sqs:
-          queueName: AutocreatedFifoQueue
+          queueName: AutocreatedFifoQueue.fifo
```
Comment (AdrieanKhisbe) [inline tests/serverless-plugins-integration/serverless.sqs.autocreate.yml]: :+1:

**Comment by AdrieanKhisbe** [comment]
@Letsch22 cool!
Thanks for the fix, and the detailled explanation about why!

I'll have an overall look on the PR now, and validate it if everything is good.

No problem if I wait for the end of week/holidays to merge it and publish?

**Comment by Letsch22** [comment]
@AdrieanKhisbe end of the week is no problem at all! Thanks for reviewing this quickly.

**Comment by AdrieanKhisbe** [comment]
@Letsch22 it's live! :)
```
Successfully published:
 - serverless-offline-sqs@3.1.0
```

**Comment by rajuashok** [comment]
Hi there! Thanks for building support for FIFO. I'm trying to use it myself right now, but am running into an issue.

The error I'm getting is: 
`Value 0e58d81d for parameter MessageGroupId is invalid. The request include parameter that is not valid for this queue type`

This indicates that the queue being created is still a standard queue, since those don't support MessageGroupId's. Here's exactly what I'm doing:

In serverless.yml define resources:
```
resources:
  - ${file(src/resources/api-gateway-errors.yml)}
  - ${file(src/resources/api-gateway-authorizer.yml)}
  - ${file(src/resources/queues.yml)}
```

in ..resources/queues.yml:
```
Resources:
  MyQueue:
    Type: "AWS::SQS::Queue"
    Properties:
      QueueName: ${self:service.name}-${self:custom.stage}-notifications.fifo
      FifoQueue: true
      ContentBasedDeduplication: true
  OrderQueue:
    Type: "AWS::SQS::Queue"
    Properties:
      QueueName: ${self:service.name}-${self:custom.stage}-order.fifo
      FifoQueue: true
      ContentBasedDeduplication: true
```

I'm using elasticmq to run the queue server: `java -jar elasticmq-server-0.15.3.jar`

When I run my serverless server locally I see the following output from elasticmq:
```
09:19:40.429 [elasticmq-akka.actor.default-dispatcher-6] INFO  o.elasticmq.actor.QueueManagerActor - Creating queue QueueData(alpha-order-queue,MillisVisibilityTimeout(30000),PT0S,PT0S,2020-01-08T09:19:40.397-08:00,2020-01-08T09:19:40.397-08:00,None,false,false,None,None,Map())
09:19:40.443 [elasticmq-akka.actor.default-dispatcher-6] INFO  o.elasticmq.actor.QueueManagerActor - Creating queue QueueData(alpha-notifications-queue,MillisVisibilityTimeout(30000),PT0S,PT0S,2020-01-08T09:19:40.397-08:00,2020-01-08T09:19:40.397-08:00,None,false,false,None,None,Map())
```

I also have serverless-offline-sqs@3.1.1 installed.

What am I doing wrong? Appreciate your help!

**Comment by rajuashok** [comment]
This is amazing, thank you for this!!!

**Pull request merged**